homeassistant/components/sensor/__init__.py

```diff
@@ -546,6 +546,87 @@ class SensorStateClass(StrEnum):
+        SensorStateClass.TOTAL_INCREASING,
+    },
+    SensorDeviceClass.DATE: {None},
+    SensorDeviceClass.DISTANCE: {None}.union(set(SensorStateClass)),
+    SensorDeviceClass.DURATION: {None},
+    SensorDeviceClass.ENERGY: {
+        None,
+        SensorStateClass.TOTAL,
+        SensorStateClass.TOTAL_INCREASING,
+    },
+    SensorDeviceClass.ENUM: {None},
+    SensorDeviceClass.FREQUENCY: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.GAS: {
+        None,
+        SensorStateClass.TOTAL,
+        SensorStateClass.TOTAL_INCREASING,
+    },
+    SensorDeviceClass.HUMIDITY: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.ILLUMINANCE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.IRRADIANCE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.MOISTURE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.MONETARY: {
+        None,
+        SensorStateClass.TOTAL,
+        SensorStateClass.TOTAL_INCREASING,
+    },
+    SensorDeviceClass.NITROGEN_DIOXIDE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.NITROGEN_MONOXIDE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.NITROUS_OXIDE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.OZONE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.PM1: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.PM10: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.PM25: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.POWER_FACTOR: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.POWER: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.PRECIPITATION: {
+        None,
+        SensorStateClass.TOTAL,
+        SensorStateClass.TOTAL_INCREASING,
+    },
+    SensorDeviceClass.PRECIPITATION_INTENSITY: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.PRESSURE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.REACTIVE_POWER: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.SIGNAL_STRENGTH: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.SOUND_PRESSURE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.SPEED: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.SULPHUR_DIOXIDE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.TEMPERATURE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.TIMESTAMP: {None},
+    SensorDeviceClass.VOLATILE_ORGANIC_COMPOUNDS: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.VOLTAGE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.VOLUME: {
+        None,
+        SensorStateClass.TOTAL,
+        SensorStateClass.TOTAL_INCREASING,
+    },
+    SensorDeviceClass.WATER: {
+        None,
+        SensorStateClass.TOTAL,
+        SensorStateClass.TOTAL_INCREASING,
+    },
+    SensorDeviceClass.WEIGHT: {
+        None,
+        SensorStateClass.TOTAL,
+        SensorStateClass.TOTAL_INCREASING,
+    },
```

Comment:
Same here, when I put my cat on a diet it's supposed to loose weight.

Reply:
I think TOTAL is fine here, this is something which makes sense to plot as either the absolute value, or the growth over a period.
I don't think TOTAL_INCREASING is useful though.

Reply:
I think the underlying question is: should "min" and "max" be allowed statistics for total? Average weight of the cat makes somewhat little sense, but min and max does.